Cython/Compiler/ExprNodes.py

```diff
@@ -8866,6 +8866,10 @@ def coerce_to(self, dst_type, env):
                         if isinstance(value, CoerceToPyTypeNode):
                             value = value.arg
                         item.value = value.coerce_to(member.type, env)
+        elif dst_type.is_cpp_class and dst_type.can_coerce_from_pyobject(env):
+            # not guaranteed that we can coerce from this particular PyObject, but worth a try
+            # Note that other places seem to try calling create_from_py_utility_code instead
+            return CoerceFromPyTypeNode(dst_type, self, env)
```

Comment:
Shouldn't we just call `ExprNode.coerce_to()` as a general fallback?
(BTW, `can_coerce_from_pyobject()` was just added later than `create_from_py_utility_code()` to avoid always doing the actual work, if all we need is know if it can be done.)

Reply:
It seems to work. It's possible there might be some tests looking for the specific error message that need accounting for... we'll see